packages/devtools/lib/src/memory/memory_service.dart

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:vm_service_lib/vm_service_lib.dart';
+
+import '../globals.dart';
+
+String get _isolateId => serviceManager.isolateManager.selectedIsolate.id;
+
+Future<InstanceRef> evaluate(String objectRef, String expression) async {
+  final dynamic result =
+      await serviceManager.service.evaluate(_isolateId, objectRef, expression);
+  switch (result.runtimeType) {
+    case InstanceRef:
+      return InstanceRef.parse(result.json);
+      break;
+    default:
+      print('ERROR: Unknown evaluate type.');
+  }
+
+  return null;
+}
+
+Future<InboundReferences> getInboundReferences(
+    String objectRef, int maxInstances) async {
+  // TODO(terry): Expose as a stream to reduce stall when querying for 1000s
+  // TODO(terry): of instances.
+  final Map params = {
+    'targetId': objectRef,
+    'limit': maxInstances,
+  };
+
+  final Response response = await serviceManager.service.callMethod(
+    '_getInboundReferences',
+    isolateId: _isolateId,
+    args: params,
+  );
+
+  return InboundReferences(response.json);
+}
+
+class InboundReferences extends Response {
+  InboundReferences(Map<String, dynamic> json) {
+    elements = json['references']
+        .map<InboundReference>((rmap) => InboundReference.parse(rmap))
+        .toList();
+  }
+
+  Iterable<InboundReference> elements;
+}
+
+class InboundReference extends Response {
+  InboundReference._fromJson(Map<String, dynamic> json) {
+    //source = json['source'],
+    parentField = createServiceObject(json['parentField']);
+    parentListIndex = json['parentListIndex'];
+    parentWordOffset = json['_parentWordOffset'];
+  }
+
+  static InboundReference parse(Map<String, dynamic> json) {
+    return json == null ? null : new InboundReference._fromJson(json);
+  }
+
+  dynamic parentField;
```

Comment:
nit: add blank lines in between these vars

Reply:
Done.